x-pack/plugins/security_solution/server/lib/entity_analytics/risk_engine/risk_engine_privileges.ts

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type {
+  IKibanaResponse,
+  KibanaRequest,
+  KibanaResponseFactory,
+  StartServicesAccessor,
+} from '@kbn/core/server';
+import { buildSiemResponse } from '@kbn/lists-plugin/server/routes';
+import type { SecurityPluginStart } from '@kbn/security-plugin/server';
+import { i18n } from '@kbn/i18n';
+import type { EntityAnalyticsPrivileges } from '../../../../common/api/entity_analytics/common';
+import type { SecuritySolutionPluginStartDependencies } from '../../../plugin_contract';
+import type { SecuritySolutionRequestHandlerContext } from '../../../types';
+import {
+  RISK_ENGINE_REQUIRED_ES_CLUSTER_PRIVILEGES,
+  RISK_ENGINE_REQUIRED_ES_INDEX_PRIVILEGES,
+  getMissingRiskEnginePrivileges,
+} from '../../../../common/entity_analytics/risk_engine';
+import { checkAndFormatPrivileges } from '../utils/check_and_format_privileges';
+
+export const getUserRiskEnginePrivileges = async (
+  request: KibanaRequest,
+  security: SecurityPluginStart
+) => {
+  return checkAndFormatPrivileges({
+    request,
+    security,
+    privilegesToCheck: {
+      elasticsearch: {
+        cluster: RISK_ENGINE_REQUIRED_ES_CLUSTER_PRIVILEGES,
+        index: RISK_ENGINE_REQUIRED_ES_INDEX_PRIVILEGES,
+      },
+    },
+  });
+};
+
+export const _getMissingPrivilegesMessage = (riskEnginePrivileges: EntityAnalyticsPrivileges) => {
```

Comment:
This just generates a pretty message for the response body, see the unit tests for example input/output

Reply:
Ah it's a pattern I've picked up (somewhere?) if you're exporting something that you don't intend to be used, in this case I've only exported it to unit test it. Happy to remove the underscore if its doing more harm than good